cmd/operator/main.go

```diff
@@ -237,3 +250,28 @@ func printHealthCriteria(logger *logrus.Logger, healthCriteria []config.HealthCr
 		lg.Debug("found health criterion")
 	}
 }
+
+func determineDefaultProject() (string, error) {
+	if metadata.OnGCE() {
+		v, err := metadata.ProjectID()
+		if err != nil {
+			return "", errors.Wrapf(err, "error when getting project ID from compute metadata: %v", err)
+		}
+		return v, nil
+	}
+
+	// Try to get project ID by retrieving default value in gcloud.
+	cmd := exec.Command("gcloud", "config", "get-value", "core/project")
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	err := cmd.Run()
+	if err != nil {
+		return "", errors.Wrap(err, "error when running gcloud command to get default project")
```

Comment:
would be good to return the error message as well. so capture stderr in another buffer